ex03.ex

```diff
@@ -55,8 +55,22 @@ defmodule Ex03 do
 
   """
 
```

Comment:
## -2 

Wow. This is an interesting way of doing this. But it is unnecessarily complex. You could just do

~~~ elixir
  def odd_even([]), do: []
  def odd_even([ h | t ]), do: [ atom_for(h) | odd_even(t) ]

  defp atom_for(n) when Integer.is_odd(n),  do:  :odd
  defp atom_for(n) when Integer.is_even(n), do:  :even
~~~